integration/src/main/java/scripts/testscripts/CloneReferencedResources.java

```diff
@@ -0,0 +1,160 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import scripts.utils.ClientTestUtils;
+import scripts.utils.CloudContextMaker;
+import scripts.utils.DataRepoTestScriptBase;
+import scripts.utils.ResourceMaker;
+import scripts.utils.WorkspaceAllocateTestScriptBase;
+
+public class CloneReferencedResources extends DataRepoTestScriptBase {
+
+  private static final String DATASET_RESOURCE_NAME = "dataset_resource_name";
+  private static final Logger logger = LoggerFactory.getLogger(CloneReferencedResources.class);
+  private static final String CLONED_BUCKET_RESOURCE_NAME = "a_new_name";
+  private static final String CLONED_DATASET_RESOURCE_NAME = "a_cloned_reference";
+  private static final String CLONED_DATASET_DESCRIPTION = "Second star to the right.";
+
+  private DataRepoSnapshotResource sourceDataRepoSnapshotReference;
+  private GcpGcsBucketResource sourceBucketReference;
+  private GcpBigQueryDatasetResource sourceBigQueryDatasetReference;
+  private UUID destinationWorkspaceId;
+  private ReferencedGcpResourceApi referencedGcpResourceApi;
+
+  @Override
+  protected void doSetup(List<TestUserSpecification> testUsers, WorkspaceApi workspaceApi)
+      throws Exception {
+    super.doSetup(testUsers, workspaceApi);
+    ApiClient apiClient = ClientTestUtils.getClientForTestUser(testUsers.get(0), server);
+    referencedGcpResourceApi = new ReferencedGcpResourceApi(apiClient);
+    String bucketReferenceName = RandomStringUtils.random(16, true, false);
+    // create reference to existing test bucket
+    sourceBucketReference =
+        ResourceMaker
+            .makeGcsBucketReference(referencedGcpResourceApi, getWorkspaceId(), bucketReferenceName);
+
+    sourceBigQueryDatasetReference = ResourceMaker.makeBigQueryReference(referencedGcpResourceApi, getWorkspaceId(),
+        DATASET_RESOURCE_NAME);
+
+    final String snapshotReferenceName = RandomStringUtils.random(6, true, false);
+
+    // Create a data repo snapshot reference
+    sourceDataRepoSnapshotReference = ResourceMaker.makeDataRepoSnapshotReference(referencedGcpResourceApi,
+        getWorkspaceId(), snapshotReferenceName, getDataRepoSnapshotId(), getDataRepoInstanceName());
+
+    // create a new workspace with cloud context
+    destinationWorkspaceId = UUID.randomUUID();
+    final var requestBody =
+        new CreateWorkspaceRequestBody()
+            .id(destinationWorkspaceId)
+            .spendProfile(getSpendProfileId())
+            .stage(getStageModel());
+    final CreatedWorkspace createdDestinationWorkspace = workspaceApi.createWorkspace(requestBody);
+    assertThat(createdDestinationWorkspace.getId(), equalTo(destinationWorkspaceId));
+
+    // create destination cloud context
+    String destinationProjectId = CloudContextMaker
+        .createGcpCloudContext(destinationWorkspaceId, workspaceApi);
+    logger.info("Created destination project {} in workspace {}", destinationProjectId, destinationWorkspaceId);
```

Comment:
If we're just cloning references, I don't think we need a cloud context.

Reply:
Awesome! Removing that takes the test time down to 30 seconds.